client/core/bookie.go

```diff
@@ -83,14 +84,14 @@ func (c *candleCache) init(in []*msgjson.Candle) {
 
 // addCandle adds the candle using candles.Cache.Add. It returns most recent candle
 // in cache.
-func (c *candleCache) addCandle(msgCandle *msgjson.Candle) (recent msgjson.Candle, ok bool) {
+func (c *candleCache) addCandle(msgCandle *msgjson.Candle) *msgjson.Candle {
 	if atomic.LoadUint32(&c.on) == 0 {
-		return msgjson.Candle{}, false
+		return nil
 	}
 	c.candleMtx.Lock()
 	defer c.candleMtx.Unlock()
 	c.Add(msgCandle)
-	return *c.Last(), true
+	return c.Last()
```

Comment:
No need to make a copy now, can optimise a bit pointer (like it previously was).

Reply:
I personally prefer to use values instead of pointers (unless the value is quite large so that copying it likely be obviously slow; it depends on many many things we don't control though, such as cache-line size on a particular processor code is running and so on) because typically pointers leave more room to shoot yourself in the foot.

For example, while using pointer in `c      chan *BookUpdate` is probably gonna work fine with the changes in this PR we might want to consider changing it to `c      chan BookUpdate`.
It's not a big deal (no need to change it here), just explaining the way I see it and wonder if you strongly disagree with this reasoning (so we are on the same page in the future).

Reply:
> or example, while using pointer in `c chan *BookUpdate` is probably gonna work fine with the changes in this PR we might want to consider changing it to `c chan BookUpdate`.

As I have explained, and you acknowledged, that does not solve any of the problems, which was they payload.
Your perspective about value vs pointer has some validity, but we've established this is not relevant.